src/utils/network/session-helper.js

```diff
@@ -405,45 +412,56 @@ export const parseMessageTTL = (messageTTLThing) => {
  * @function buildMessageTTL
  * @param {Session} session - Solid's Session Object (see {@link Session})
  * @param {Date} date - JavaScript Date object
- * @param {string} senderName - Name of sender
- * @param {string} recipientName - Name of recipient
- * @param {URL} recipientWebId - webId of recipient
- * @param {Thing[]} messageTTLThing - List of message Things from message boxes
+ * @param {object} messageObject - Object containing information for the message
+ * content
+ * @param {object} messageMetadata - Object containing information about the message
+ * @param {string} buildFor - String for "Sender" or "Recipient"
  * @returns {object} messageObject - An object containinng the message content,
  * title, uploadDate, sender, and recipient
  */
-
-// TODO: potentially add other useful fields - see https://github.com/codeforpdx/PASS/pull/191#discussion_r1217249834
-
-export const buildMessageTTL = (
-  session,
-  date,
-  messageObject,
-  senderName,
-  recipientName,
-  recipientWebId
-) => {
+export const buildMessageTTL = (session, date, messageObject, messageMetadata, buildFor) => {
   const newMessageTTL = buildThing(createThing({ name: 'message' }))
     .addDatetime(RDF_PREDICATES.uploadDate, date)
     .addStringNoLocale(RDF_PREDICATES.title, messageObject.title)
     .addStringNoLocale(RDF_PREDICATES.message, messageObject.message)
     .build();
 
+  const newMessageID = buildThing(createThing({ name: 'messageid' }))
+    .addStringNoLocale(RDF_PREDICATES.identifier, messageMetadata.messageId)
+    .addUrl(
+      RDF_PREDICATES.url,
+      buildFor === 'sender'
+        ? `${messageMetadata.podUrl}PASS/Outbox/${messageMetadata.messageSlug}.ttl`
+        : `${messageMetadata.recipientPodUrl}PASS/Inbox/${messageMetadata.messageSlug}.ttl`
+    )
+    .build();
+
   const senderInfo = buildThing(createThing({ name: 'sender' }))
-    .addStringNoLocale(RDF_PREDICATES.sender, senderName)
+    .addStringNoLocale(RDF_PREDICATES.sender, messageMetadata.senderName)
     .addUrl(RDF_PREDICATES.url, session.info.webId)
     .build();
 
   const recipientInfo = buildThing(createThing({ name: 'recipient' }))
-    .addStringNoLocale(RDF_PREDICATES.recipient, recipientName)
-    .addUrl(RDF_PREDICATES.url, recipientWebId)
+    .addStringNoLocale(RDF_PREDICATES.recipient, messageMetadata.recipientName)
+    .addUrl(RDF_PREDICATES.url, messageMetadata.recipientWebId)
     .build();
 
   let newSolidDataset = createSolidDataset();
-  [newMessageTTL, senderInfo, recipientInfo].forEach((thing) => {
+  [newMessageTTL, newMessageID, senderInfo, recipientInfo].forEach((thing) => {
     newSolidDataset = setThing(newSolidDataset, thing);
   });
 
+  let replyTo;
+
+  if (messageObject.inReplyTo) {
+    replyTo = buildThing(createThing({ name: 'replyTo' }))
+      .addStringNoLocale(RDF_PREDICATES.identifier, messageObject.inReplyTo)
+      .addUrl(RDF_PREDICATES.url, messageObject.messageUrl)
```

Comment:
Depending on sender or recipient, this is not correctly giving the right url to the previous message. Needs to be updated

Reply:
Got this fixed.

Reply:
Got this fixed.